fix: breakage with Node v9.4.0.

jinwoo opened the pull request:
`pushStream()`'s callback has changed from `(pushedStream, headers) =>
{}` to `(err, pushedStream, headers) => {}` with Node v9.4.0.

jinwoo commented:
I'm not sure why codecov says there's no coverage change. Looking.

jinwoo commented:
The `index.ts` file is not included in the coverage data from `nyc`. I'm not sure why. I'll investigate it separately. PTAL.

jinwoo commented:
Changed `?:` to `if-else`, and that makes `index.ts` included in the coverage data.

@kjin Can you take a look?

ofrobots commented on ts/src/index.ts:
```diff
@@ -117,12 +121,26 @@ export class AutoPush {
+        };
+        // Node 9.4.0 changed the callback function signature.
+        let callback: PushStreamCallback;
+        if (isNode940) {
```
I expect that the offending change from 9.4.0 is going to be backported to Node 8 as well. For that version of Node 8 this code will break.

An alternative to having a semver check in the code would be to use a `engines` field in package.json that requires Node 9.4+ or Node 8.10+ (not yet released)

jinwoo replied:
Oh, is it true that the same change will be backported to 8? If we limit the engines to 9.4+ and 8.10+, then it'll be broken with current Node 8...

jinwoo replied:
Does the "engines" field actually work as expected?

I added
```json
  "engines": {
    "node": ">=9.4.0 || >8.9.4 <9"
  },
```

But when I do `npm install -g` with Node 8.9.4, it doesn't emit any warnings.

ofrobots replied:
What's that <9 in that expression, not sure if I understand the expression.

jinwoo replied:
"version >= 9.4.0 OR (version > 8.9.4 AND < 9.0.0)"

Am I missing something?

ofrobots replied:
Ah, whitespace does AND.

ofrobots replied:
Not sure why it doesn't warn though.

ofrobots replied:
https://runkit.com/embed/7g1z4mkyclcy

jinwoo commented:
Added the "engines" field to package.json and deleted the conditional callback creation.

Also temporarily disabled CircleCI tests on Node 8.

PTAL
